src/main/java/by/tms/insta/service/UserService.java

```diff
@@ -0,0 +1,53 @@
+package by.tms.insta.service;
+
+import by.tms.insta.dao.UserStorage;
+import by.tms.insta.entity.User;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.List;
+
+/**
+ * UserService
+ *
+ * @author Stepan Vasilyeu
+ * @since 30.05.2022
+ */
+public class UserService {
+    private final UserStorage userStorage;
+    private final UserValidator userValidator;
+
+    public UserService(@Qualifier("hibernateUserStorage") UserStorage userStorage, UserValidator userValidator) {
+        this.userStorage = userStorage;
+        this.userValidator = userValidator;
+    }
+
+    public boolean save(User user) {
+        if (!userValidator.loginAndNameValidate(user.login, user.name) || !userValidator.passwordValidate(user.password)) {
+            return false;
+        }
+        if (findUserByLogin(user.login) != null) {
+            return false;
+        }
```

Comment:
You should use UserExists instead of this, because it will cause NullPointerException before of comparing operation